### PR TITLE
Make NetworkingModule extensible so we can modify requests

### PR DESCRIPTION
**Pull request by felipecsl**
## Motivation

This allows us to have the chance to modify an OkHttp request before sending.
I broke down `NetworkingModule#sendRequest()` into multiple smaller methods that can be either overridden in subclasses or called individually from a separate class.

## Test Plan

Manual testing pretty much. No functionality should change.

### CODE REVIEW

File: ReactAndroid/src/main/java/com/facebook/react/modules/network/ResponseCallback.java [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+import java.io.IOException;
+import java.io.Reader;
+
+import okhttp3.Call;
+import okhttp3.Callback;
+import okhttp3.Headers;
+import okhttp3.Response;
+import okhttp3.ResponseBody;
+
+public class ResponseCallback implements Callback {
```
Comment (felipecsl) [inline ReactAndroid/src/main/java/com/facebook/react/modules/network/ResponseCallback.java]: This stuff was just moved out of `NetworkingModule`

File: ReactAndroid/src/main/java/com/facebook/react/modules/network/NetworkingModule.java [hunk 2]
```diff
@@ -45,13 +39,13 @@
...
 import okhttp3.RequestBody;
 import okhttp3.Response;
-import okhttp3.ResponseBody;
 
 /**
  * Implements the XMLHttpRequest JavaScript interface.
  */
 @ReactModule(name = "RCTNetworking", supportsWebWorkers = true)
-public final class NetworkingModule extends ReactContextBaseJavaModule {
+public final class NetworkingModule extends ReactContextBaseJavaModule implements
```
Comment (gpeal) [inline ReactAndroid/src/main/java/com/facebook/react/modules/network/NetworkingModule.java]: @felipecsl Do we actually want to reformat Facebook's code? It makes it harder to see the diff if we ever need to.

Comment (gpeal) [reply ReactAndroid/src/main/java/com/facebook/react/modules/network/NetworkingModule.java]: @felipecsl Can you also add docs for what we've changed from the original?

**Review by gpeal — COMMENTED** [review]
What is your code vs Facebook's?

**Comment by lelandrichardson** [comment]
@felipecsl i agree with gabe, let's try and minimize the diff here if we can. It looks like there's some changes in here that are just aesthetic that I'd prefer we avoid.

File: ReactAndroid/src/main/java/com/facebook/react/modules/network/NetworkingModule.java (hunk 2 above)
Comment (felipecsl) [reply ReactAndroid/src/main/java/com/facebook/react/modules/network/NetworkingModule.java]: the android studio auto-reformat makes my life harder. I'll try to revert the formatting changes.
As for this new interface, I can probably remove it

**Review by felipecsl — COMMENTED** [review]
Agreed, I'll revert the unrelated stuff

**Comment by felipecsl** [comment]
Alright I think i've reverted everything except just the absolutely necessary changes. PTAL @lelandrichardson @gpeal

File: ReactAndroid/src/main/java/com/facebook/react/modules/network/NetworkingModule.java [hunk 3]
```diff
@@ -74,7 +66,7 @@
       @Nullable String defaultUserAgent,
       OkHttpClient client,
       @Nullable List<NetworkInterceptorCreator> networkInterceptorCreators) {
-    super(reactContext);
+      super(reactContext);
```
Comment (gpeal) [inline ReactAndroid/src/main/java/com/facebook/react/modules/network/NetworkingModule.java]: This was right before

**Comment by felipecsl** [comment]
image
good to go

File: ReactAndroid/src/main/java/com/facebook/react/modules/network/ResponseCallback.java (hunk 1 above)
Comment (gpeal) [inline ReactAndroid/src/main/java/com/facebook/react/modules/network/ResponseCallback.java]: javadoc? Will be helpful if you plan to upstream this.

**Review by gpeal — APPROVED** [review]
I would suggest adding javadocs which will also help when you upstream it. Other than that, LGTM

**Pull request merged**